openapi-validator/src/main/java/io/ballerina/openapi/validator/ResourceWithOperation.java

```diff
@@ -288,37 +291,45 @@
             NodeList<Node> nodes = functionDefinitionNode.relativeResourcePath();
             String functionMethod = functionDefinitionNode.functionName().text().trim();
             Map<String, Node> parameterNodeMap = new HashMap<>();
-
-            StringBuilder stringBuilder = new StringBuilder();
-            String functionPath = "/";
-            stringBuilder.append(functionPath);
-            Iterator<Node> pathNodeIter = nodes.iterator();
-            while (pathNodeIter.hasNext()) {
-                Node next = pathNodeIter.next();
-                String node = next.toString().trim();
-                if (next instanceof ResourcePathParameterNode) {
-                    ResourcePathParameterNode pathParameterNode = (ResourcePathParameterNode) next;
-                    String paramName = pathParameterNode.paramName().text().trim();
-                    node = "{" + paramName + "}";
-                    parameterNodeMap.put(paramName, next);
-                }
-                stringBuilder.append(node);
-            }
-            if (resourceSummaryList.containsKey(stringBuilder.toString())) {
+            String path = generatePath(nodes, parameterNodeMap);
+            if (resourceSummaryList.containsKey(path)) {
                 setParametersToMethod(functionDefinitionNode, functionMethod, parameterNodeMap,
-                        resourceSummaryList.get(stringBuilder.toString()));
+                        resourceSummaryList.get(path));
             } else {
                 ResourcePathSummary resourcePathSummary = new ResourcePathSummary();
                 //Set location as first function location
-                resourcePathSummary.setPath(stringBuilder.toString());
+                resourcePathSummary.setPath(path);
                 resourcePathSummary.setPathPosition(functionDefinitionNode.location());
                 setParametersToMethod(functionDefinitionNode, functionMethod, parameterNodeMap, resourcePathSummary);
-                resourceSummaryList.put(stringBuilder.toString(), resourcePathSummary);
+                resourceSummaryList.put(path, resourcePathSummary);
             }
         }
         return resourceSummaryList;
     }
 
+    /**
+     * This function generates resource function path that align to OAS format.
+     */
+    private static String generatePath(NodeList<Node> nodes, Map<String, Node> parameterNodeMap) {
+        StringBuilder pathBuilder = new StringBuilder();
+        pathBuilder.append(SLASH);
```

Comment:
Yeah, This won't be affected to the OS-specific since we take details from directly OpenAPI contract to compare with syntax tree and OAS allows to user to document the path with forwarding slash `/` only. Here what we do is traverse the syntax node for generating path which aligns to the OpenAPI path format. 🙂